src/main/java/seedu/address/logic/parser/CheckoutCommandParser.java

```diff
@@ -17,8 +22,24 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public CheckoutCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_ROOM, PREFIX_DATE_START, PREFIX_DATE_END);
+        if (!argMultimap.getValue(PREFIX_ROOM).isPresent()
+                || !argMultimap.getPreamble().isEmpty()
+                || (argMultimap.getValue(PREFIX_DATE_START).isPresent()
+                    && !argMultimap.getValue(PREFIX_DATE_END).isPresent())
+                || (argMultimap.getValue(PREFIX_DATE_END).isPresent()
+                    && !argMultimap.getValue(PREFIX_DATE_START).isPresent())) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckoutCommand.MESSAGE_USAGE));
+        }
         try {
-            RoomNumber roomNumber = ParserUtil.parseRoomNumber(args);
+            RoomNumber roomNumber = ParserUtil.parseRoomNumber(argMultimap.getValue(PREFIX_ROOM).get());
+            if (ParserUtil.arePrefixesPresent(argMultimap, PREFIX_DATE_START, PREFIX_DATE_END)) {
```

Comment:
Would be more readable if this was the 'else' block instead!

Reply:
How would it be more readable? It's a matter of personal preference, I believe.

Reply:
Not an important point, but it follows what is taught in the module, see 

- Code readability: 'Tripping Up', Make the Happy Path Prominent
- Tutorial 7, third block